src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -0,0 +1,214 @@
+        $contentType = $content->getContentType();
+
+        $numberOfSubitemsToLoad = $this->resolveLoadLimit($loadSubtreeRequestNode);
+
+        $children = [];
+        if ($depth < $this->maxDepth && $loadChildren) {
+            $searchResult = $this->findSubitems($location, $numberOfSubitemsToLoad);
+            $totalChildrenCount = $searchResult->totalCount;
+
+            /** @var \eZ\Publish\API\Repository\Values\Content\Location $childLocation */
+            foreach (array_column($searchResult->searchHits, 'valueObject') as $childLocation) {
+                $childLoadSubtreeRequestNode = null !== $loadSubtreeRequestNode
+                    ? $this->findChild($childLocation->id, $loadSubtreeRequestNode)
+                    : null;
+
+                $children[] = $this->createNode(
+                    $childLocation,
+                    $childLoadSubtreeRequestNode,
+                    null !== $childLoadSubtreeRequestNode,
+                    $depth++
+                );
+            }
+        } else {
+            $totalChildrenCount = $this->countSubitems($location);
+        }
+
+        return new Node(
+            $depth,
+            $location->id,
+            $location->contentId,
+            $content->getName(),
+            $contentType->identifier,
+            $contentType->isContainer,
+            $location->invisible,
+            $totalChildrenCount,
+            $numberOfSubitemsToLoad,
+            $children
+        );
+    }
+
+    /**
+     * @param \EzSystems\EzPlatformAdminUi\REST\Value\ContentTree\LoadSubtreeRequestNode|null $loadSubtreeRequestNode
+     *
+     * @return int
+     */
+    private function resolveLoadLimit(?LoadSubtreeRequestNode $loadSubtreeRequestNode): int
+    {
+        $limit = $this->displayLimit;
+
+        if (null !== $loadSubtreeRequestNode) {
+            $limit = $loadSubtreeRequestNode->limit;
+        }
+
+        if ($limit > $this->childrenLoadMaxLimit) {
+            $limit = $this->childrenLoadMaxLimit;
+        }
+
+        return $limit;
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $parentLocation
+     * @param int $limit
+     * @param int $offset
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Search\SearchResult
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function findSubitems(Location $parentLocation, int $limit = 10, int $offset = 0): SearchResult
+    {
+        $searchQuery = $this->getSearchQuery($parentLocation);
+
+        $searchQuery->limit = $limit;
+        $searchQuery->offset = $offset;
+
+        return $this->searchService->findLocations($searchQuery);
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $parentLocation
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\LocationQuery
+     */
+    private function getSearchQuery(Location $parentLocation): LocationQuery
```

Comment:
Yes/no.

It always great if you can extend something - I agree. The problem is - we do not have similar extension point on UDW and Subitems which means you will see different results there. Another reason we I'd not add an extension point in CT is to not sacrifice performance which is our biggest concern.

Reply:
Well, this could be always done as follow up story and isn't a blocker  to merge for me.

What I had in mind here as extension point is rather simple and shouldn't hit performance: allow developers to modify excluding criterion by subscribing event or maybe reuse   `\eZ\Publish\Core\QueryType\QueryType` for that.